lib/spack/spack/test/url_fetch.py

```diff
@@ -40,8 +48,16 @@ def fn(v):
             main_url = url or urls[0]
             return spack.url.substitute_version(main_url, v)
 
+        def fn_urls(v):
+            urls_loc = urls or [url]
+            return [spack.url.substitute_version(main_url, v) for main_url in urls_loc]
```

Comment:
(minor) IMO it's confusing to use `main_url` as the iteration variable name since I thought only the first URL was the "main" one. In other words: what does "main" mean here?

Reply:
It means "I copy and pasted it," I'll make it `u` or something.